arangod/RocksDBEngine/RocksDBDumpContext.cpp

```diff
@@ -238,14 +274,23 @@ RocksDBDumpContext::RocksDBDumpContext(RocksDBEngine& engine,
     _threads.emplace_back(
         [this, guard = BoundedChannelProducerGuard(_channel)] {
           try {
-            while (true) {
+            while (!stopped()) {
               // will block until all workers wait, i.e. no work is left
               auto workItem = _workItems.pop();
               if (workItem.empty()) {
                 break;
               }
 
-              handleWorkItem(std::move(workItem));
+              do {
+                try {
+                  handleWorkItem(workItem);
+                  break;
+                } catch (basics::Exception const& ex) {
+                  if (ex.code() != TRI_ERROR_RESOURCE_LIMIT) {
+                    throw ex;
+                  }
+                }
+              } while (!stopped());
```

Comment:
maybe some limit for iteration count here?

Reply:
You are right, there is currently no limit on the number of iterations.
However, the iterations will effectively be bounded because of the `!stopped()` condition.
Each dump context has a TTL, and if no new requests come in for a dump context (e.g. because all threads are stalled), then it will eventually time out and `stopped()` will become true.
Then all threads that cannot make any progress will be stopped.